index.test-d.ts

```diff
@@ -0,0 +1,150 @@
+import {expectError, expectType, printType} from 'tsd';
+import pify from '.';
+
+expectError(pify());
+expectError(pify(null));
+expectError(pify(undefined));
+expectError(pify(123));
+expectError(pify('abc'));
+expectError(pify(null, {}));
+expectError(pify(undefined, {}));
+expectError(pify(123, {}));
+expectError(pify('abc', {}));
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+expectType<never>(pify((v: number) => {})());
+expectType<never>(pify(() => 'hello')());
```

Comment:
I'm wondering if these should actually produce `Promise<unknown>` 🤔 

Neither of these result in a runtime error and as we can't make this an error at the callsite I think matching the runtime behaviour makes more sense.

What do you think?

Reply:
Done in https://github.com/tom-sherman/pify/commit/479654c179fc94e5ca0e763129965c36d635af8c